src/main/java/com/microsoft/azure/functions/ExecutionContext.java

```diff
@@ -50,5 +52,19 @@ default TraceContext getTraceContext() {
     default RetryContext getRetryContext() {
         return null;
     }
+
+    Map<String, Parameter> getParameterMap();
+
+    Map<String, String> getParameterPayloadMap();
+
+    void setMiddlewareInput(String key, Object value);
+
+    Object getReturnValue();
+
+    void setMiddlewareOutput(Object value);
+
+    void setFunctionInstance(Object functionInstance);
```

Comment:
Might consider moving these middleware specific methods like setFunctionInstance to a separate interface that extends extends ExecutionContext.  ExecutionContext is available to the application developer and these methods might not be something you want exposed to them.  See my PR here which shows this:

https://github.com/Azure/azure-functions-java-core-library/pull/10

Reply:
Thank you for your valuable suggestion!